cmd/migrate/migrate.go

```diff
@@ -78,6 +85,22 @@ func runMigration(_ *cobra.Command, _ []string) error {
 		return fmt.Errorf("unknown datastore engine type: %s", engine)
 	}
 
+	// Parse the database uri with url.Parse() and update username/password, if set via flags
+	dbURI, err := url.Parse(uri)
```

Comment:
This is specific to the postgres implementation, and does not work for the MySQL variant.

See https://github.com/openfga/openfga/blob/main/pkg/storage/postgres/postgres.go#L44-L68
vs
https://github.com/openfga/openfga/blob/main/pkg/storage/mysql/mysql.go#L42-L56

Reply:
my bad, assumed that mysql would use the same format string, just with mysql:// prefix. 

Moved the code to the datastore specific part and added a case handler for mysql as well.